cmake/BLTMacros.cmake

```diff
@@ -370,24 +378,33 @@ endmacro(blt_register_library)
 
 
 ##------------------------------------------------------------------------------
-## blt_add_library( NAME <libname>
-##                  SOURCES [source1 [source2 ...]]
-##                  HEADERS [header1 [header2 ...]]
-##                  INCLUDES [dir1 [dir2 ...]]
-##                  DEFINES [define1 [define2 ...]]
-##                  DEPENDS_ON [dep1 ...] 
-##                  OUTPUT_NAME [name]
-##                  OUTPUT_DIR [dir]
-##                  SHARED [TRUE | FALSE]
+## blt_add_library( NAME         <libname>
+##                  SOURCES      [source1 [source2 ...]]
+##                  HEADERS      [header1 [header2 ...]]
+##                  INCLUDES     [dir1 [dir2 ...]]
+##                  DEFINES      [define1 [define2 ...]]
+##                  DEPENDS_ON   [dep1 ...] 
+##                  OUTPUT_NAME  [name]
+##                  OUTPUT_DIR   [dir]
+##                  SHARED       [TRUE | FALSE]
+##                  OBJECT       [TRUE | FALSE]
 ##                  CLEAR_PREFIX [TRUE | FALSE]
-##                  FOLDER [name]
+##                  FOLDER       [name]
 ##                 )
 ##
 ## Adds a library target, called <libname>, to be built from the given sources.
 ## This macro uses the BUILD_SHARED_LIBS, which is defaulted to OFF, to determine
 ## whether the library will be build as shared or static. The optional boolean
 ## SHARED argument can be used to override this choice.
 ##
+## The OBJECT argument creates a CMake object library. Basically it is a collection
+## of compiled source files that are not archived or linked. Unlike regular CMake
+## object libraries you do not have to use the $<TARGET_OBJECTS:<libname>> syntax,
+## you can just use <libname>.
+##    Note: Object libraries do not follow CMake's transitivity rules until 3.13.
```

Comment:
Forgot to mention -- can we add a check/warning for usage of object libraries if the cmake version is too low?  
This will give users an explanation for unexpected behavior, and make it more obvious that they should upgrade if they want to depend on this capability.

Reply:
@kennyweiss I think for some cases it will work just fine, but the transitive case won't be flushed out until folks use newer cmake. Not sure a warning is necessary b/c it's subtle. 

For example, I think I can use this now b/c in a few cases where transitive wont be a requirement.

Reply:
The more I think about this the more I think transitivity is not a problem.  This utilized the registered library logic and will add info from all the dependencies.  I don’t think you can create a ordering problem with object libraries because they aren’t linked.